### PR TITLE
Make access to self immutable where possible

### PR DESCRIPTION
**Pull request by stevepryde**
This is intended firstly as a discussion point w.r.t. issue #153 and also to show how this would look with all of the mutability requirements minimized. 

The one method where I think capturing self might be warranted is `Client::close()`.

Perhaps its sig could be changed to:
```rust
pub async fn close(self) -> Result<Client, error::CmdError> {
```

Obviously all of this is a highly disruptive breaking change with many implications for calling code.

### CODE REVIEW

**Comment by stevepryde** [comment]
resolves #153

File: src/client.rs [hunk 1]
```diff
@@ -812,10 +808,7 @@ impl Client {
     /// Screenshot](https://www.w3.org/TR/webdriver1/#dfn-take-element-screenshot) of the WebDriver
     /// standard.
     #[cfg_attr(docsrs, doc(alias = "Take Element Screenshot"))]
-    pub async fn screenshot_element(
-        &mut self,
-        element: Element,
-    ) -> Result<Vec<u8>, error::CmdError> {
+    pub async fn screenshot_element(&self, element: Element) -> Result<Vec<u8>, error::CmdError> {
```
Comment (jonhoo) [inline src/client.rs:811]: Seeing this, I wonder if this should be a method on `Element` instead? Incidental to this PR though.

File: src/client.rs [hunk 2]
```diff
@@ -885,7 +878,7 @@ impl Client {
             // TODO: get rid of this clone
             let current = current.clone();
             // TODO: and this one too
-            let mut c = c.clone();
+            let c = c.clone();
```
Comment (jonhoo) [inline src/client.rs:881]: This clone _may_ no longer be necessary.

**Review by jonhoo — CHANGES_REQUESTED** [review]
Part of the thinking for `&mut self` and requiring explicit clones was also to try to prevent users from shooting themselves in the foot by, say, having two threads writing into an element at the same time, but given the existence of `Clone` it's really wishful thinking anyway. So I'm on board with this change.

I like the idea of making `close` consume `self` — :+1: subject to that.

File: src/client.rs (hunk 2 above)
Comment (stevepryde) [reply src/client.rs:881]: Apparently it still is, due to lifetimes

File: src/client.rs (hunk 1 above)
Comment (stevepryde) [reply src/client.rs:811]: Probably. I'll move it (since we're already breaking everything 😅 )

**Review by jonhoo — APPROVED** [review]

**Pull request merged**